packages/perseus/src/widgets/interactive-graphs/graphs/polygon.tsx

```diff
@@ -103,11 +104,15 @@ export const PolygonGraph = (props: Props) => {
                     strokeWidth: TARGET_SIZE,
                     style: {
                         cursor: dragging ? "grabbing" : "grab",
+                        fill: hovered ? "var(--mafs-blue)" : "transparent",
                     },
                     onFocus: () => setFocused(true),
                     onBlur: () => setFocused(false),
                     onMouseEnter: () => setHovered(true),
-                    onMouseLeave: () => setHovered(false),
+                    onMouseLeave: () => {
+                        setHovered(false);
+                        setFocused(false);
```

Comment:
Why do we `setFocused(false)` here? If the user has focused the polygon with the keyboard, we don't want mousing over it and then away from it to make it appear unfocused.

Reply:
When testing, if I move the polygon, the lines stay weighted even after moving the shape. This isn't how other Mafs graphs act, so I was trying to get it to act the same. First I made it so the weighting only happens when dragging and on hover and not on focus, but that wasn't quite right. Looks like this isn't quite right either 🤔 Not sure how to fix that, but I can try to brainstorm another fix tomorrow too. Or just make a new ticket.

Reply:
Oh, that's because we're managing the focus styling ourselves, based on `onFocus`. The issue is that an element can be _focused_ but not have _visible focus_, which is the state where the browser applies focus styling to it by default. The behavior you're seeing happens because whenever you click an element, it gets _focus_, and `onFocus` is called. However, an element generally doesn't get _visible focus_ unless you select it with the keyboard. I say "generally" because I seem to recall that browsers have a lot of complicated logic around exactly when to show focus. [Docs.](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible)

I think in order to get this to work the way we want, we would either need to:

- manage the hover/focus visuals entirely in CSS, using `:hover` and `:focus-visible` pseudo-classes.
- detect `:focus-visible` by holding a ref to the polygon element and, when `onFocus` is called, do something like this:
  ```ts
  if (polygonRef.current.matches(":focus-visible")) {
    setFocused(true);
  }
  ```
  Ideally, there would be a separate event callback we could register like `onFocusVisible`, but I don't think that exists.

Reply:
After chatting with Caitlyn, we realized that polygon should not actually have its lines weighted in any way, because that is the same behavior as angle, sinusoid, and circle (which are more similar to this graph than the linear ones). The area that is interactive for the user is the center, upon which the user can grab and drag. So the lines themselves are not what are interactive. Updated the code to reflect this.